crates/bevy_ecs/src/bundle/mod.rs

```diff
@@ -244,7 +244,13 @@ pub trait DynamicBundle {
     /// Calls `func` on each value, in the order of this bundle's [`Component`]s. This passes
     /// ownership of the component values to `func`.
     #[doc(hidden)]
-    fn get_components(self, func: &mut impl FnMut(StorageType, OwningPtr<'_>)) -> Self::Effect;
+    unsafe fn get_components(ptr: *const Self, func: &mut impl FnMut(StorageType, OwningPtr<'_>));
+
+    // SAFETY:
+    // - Must be called exactly once after `get_components` has been called.
+    // - `ptr` must point to a valid instance of `Self` but does not necessary need to be aligned.
+    #[doc(hidden)]
+    unsafe fn apply_effect(ptr: *const Self, entity: &mut EntityWorldMut);
```

Comment:
Is there a reason why `ptr` can be non-aligned here, but nothing about this is stated over `get_components`? Especially considering that the `impl` for `C: Component` of `get_components` require `ptr` to be aligned in order to be sound.

I also don't see a place which can call this method with an unaligned pointer. It seems the idea was to avoid moving commands out of the `CommandQueue`, but this is not being done in PR?

Reply:
This is an artifact of being based on #20593. I'll go ahead and fix this comment.